libs/net/models/Extensions/IngestModelExtensions.cs

```diff
@@ -48,14 +49,17 @@ public static string GetConfigurationValue(this IngestModel ingest, string key,
 
         if (value is JsonElement element)
         {
-            return element.ValueKind switch
-            {
-                JsonValueKind.String => (T)Convert.ChangeType($"{element.GetString()}".Trim(), typeof(T)),
-                JsonValueKind.Null or JsonValueKind.Undefined => default,
-                JsonValueKind.Number => (T)Convert.ChangeType($"{element.GetInt32()}", typeof(T)),
-                JsonValueKind.True or JsonValueKind.False => (T)Convert.ChangeType($"{element.GetBoolean()}", typeof(T)),
-                _ => (T)Convert.ChangeType($"{element}", typeof(T)),
-            };
+            if (typeof(T).GetTypeInfo().IsEnum)
+                return (T)Enum.Parse(typeof(T), $"{element.GetString()}".Trim());
```

Comment:
This will only work if the value being parsed is a string.  You should also check the element.ValueKind.

Reply:
added check for element.ValueKind